Processor.php

```diff
@@ -84,13 +93,54 @@ public static function getMetricConditions()
         );
     }
 
+    /**
+     * @param $period
+     * @param $idSite
+     * @return void
+     * @throws RetryableException The exception that should be caught and handled by the Scheduler class. Only happens
+     * if there's an alert that can't be processed yet and needs to be retried.
+     */
     public function processAlerts($period, $idSite)
     {
         $alerts = $this->getAllAlerts($period);
 
+        $retryCount = $this->getRetryCountForCurrentTask();
```

Comment:
we have this for all the alerts in the period and not for each alerts, is that okay ?

Reply:
It should work fine since the alerts are processed in a batch by a single task for each period. The retry count is for the task, not the individual alert.

Reply:
so my Q is if we have 3 tasks and only 1 fails, will it retry all the 3 ?

Reply:
If the individual tasks (day, week, month) shouldn't affect each other. For example, if a day alert can't process, only the day task is rescheduled. Within the day alert, if alerts 1, 2, 4, 5 trigger and alert 3 fails, it will reschedule a retry. When the retry processes, alerts 1, 2, 4, 5 will be skipped and it will only try to process alert 3 again. That's what [these lines of code](https://github.com/matomo-org/plugin-CustomAlerts/pull/183/files#diff-50a4d369c4144ab2a656f379c09bac4f6b73acde762b589a50b8430effb94be5R111-R115) are about.